File: Visual-Studio-Code/Python/Python-Programs/SelectionSort.py
#Selection Sort

def swap(list,one,two):

    #You dont fucking need this in python

    temp = list[one]
    list[one] = list[two]
    list[two] = temp


def selectionSort(Arr,N):

    iMin = 0

    for i in range(0,N):

        iMin = i

        for j in range(i,N):

            if Arr[j] < Arr[iMin]:
                iMin = j

        if i != iMin:
            Arr[i],Arr[iMin] = Arr[iMin],Arr[i]


num = [7,4,2,6,3,1,5]
size = len(num)

selectionSort(num,size)

print("Sorted Array : ")
for i in range (0,size):
    print(f" {num[i]}",end="")

File: Semester-2/C/Asdos/11_Sort/selectionSort.c
#include <stdio.h>

void swap(int *one,int *two){   
    int temp = *one;
    *one = *two;
    *two = temp;
}

int selectionSort(int *arr,int size){

    for(int i=0; i<size ;i++){

        int iMin = i;

        for(int j=i; j<size ;j++){
            if(arr[j] < arr[iMin]){
                iMin = j;
            }
        }

        if(iMin != i){
            swap(&arr[iMin],&arr[i]);
        }

    }

    return 0;
}

int main(){

    int arr[] = {9,3,6,7,1,4,2,5,8};
    int size = sizeof(arr)/sizeof(arr[1]);

    selectionSort(arr,size);

    for(int i=0; i<size ;i++){
        printf("%d ",arr[i]);
    }

    return 0;
}